docs/quality_assurance.md

```diff
@@ -0,0 +1,157 @@
+
+## Interchain Security Protocol Correctness
+
+The main concern addressed in this section is the correctness of the Interchain Security protocol. In other words, the implementation should be aligned with the Interchain Security [specification](https://github.com/cosmos/ibc/blob/master/spec/appics-028-cross-chain-validation/README.md). 
+
+The implementation MUST guarantee the *Channel Uniqueness* property, i.e., the channel between the provider chain and a consumer chain MUST be unique.
+
+In addition, the implementation MUST guarantee the following [system properties](https://github.com/cosmos/ibc/blob/master/spec/appics-028-cross-chain-validation/system_model_and_properties.md#system-properties):
+- *Validator Set Replication*
+- *Bond-Based Consumer Voting Power*
+- *Slashable Consumer Misbehavior*
+- *Consumer Rewards Distribution*
+
+---
+
+| Concern re. *Channel Uniqueness* | Code Review | Unit Testing | Diff. testing | Testnet | Protocol audit |
+| ------- | ----------- | ------------ | ------------- | ------- | ----- |
+| `SpawnConsumerChainProposal(chainId)` should fail if a consumer with `chainId` already exists | `TODO` | `??` | `??` | `TODO` | `TODO` |
+| The channel handshake for a consumer with `chainId` should fail if there is already an established CCV channel for `chainId`  | `TODO` | `??` | `??` | `TODO` | `TODO` |
+| *Channel Uniqueness* should hold even if a consumer chain restarts | `TODO` | `??` | `??` | `TODO` | `TODO` |
+| *Channel Uniqueness* should hold even when a client expires | `TODO` | `??` | `??` | `TODO` | `TODO` |
+
+---
+
+| Concern re. *Validator Set Replication* | Code Review | Unit Testing | Diff. testing | Testnet | Protocol audit |
+| ------- | ----------- | ------------ | ------------- | ------- | ----- |
+| Every validator set on any consumer chain MUST either be or have been a validator set on the provider chain. | `TODO` | `NA` | `??` | `TODO` | `TODO` |
+| Every consumer chain receives the same sequence of `ValidatorSetChangePacket`s in the same order. | `TODO` | `NA` | `??` | `TODO` | `TODO` |
+
+---
+
+| Concern re. *Bond-Based Consumer Voting Power* | Code Review | Unit Testing | Diff. testing | Testnet | Protocol audit |
+| ------- | ----------- | ------------ | ------------- | ------- | ----- |
+| A power increase of a validator `val` on a consumer chain can be only due to <br /> - a `Delegate()` / `Redelegate()` to `val` on provider <br /> - `val` joining the provider validator set (another validator leaving the set) | `TODO` | `TODO` | `??` | `TODO` | `TODO` |
+| A power decrease of a validator `val` on a consumer chain can be only due to  <br /> - an `Undelegate()` / `Redelegate()` from `val` on provider <br /> - `val` gets slash on the provider chain <br /> - another validator joining the provider validator set (`val` leaving the set) | `TODO` | `TODO` | `??` | `TODO` | `TODO` |
```

Comment:
I think we should reword these two concerns. It's not clear exactly what the issue is and they aren't really testable.

Maybe something like

- 'A positive power for validator `val` on a consumer chain at height `h1` is always corresponds to an equal positive power on the provider at height `h0`, `h0 < h1` under the `<` imposed by packet delivery.

ect

But even this is getting into the weeds. Should just state the need in plain English here and link to the fully fleshed property the spec?

Reply:
What about https://github.com/cosmos/interchain-security/pull/215/commits/528262b34fcff141cdc4e2483b289901e871f9a3?

Reply:
Yes, it's much clearer, nice 👍